x/collection/msgs.go

```diff
@@ -166,26 +166,26 @@ func ValidatePermission(permission Permission) error {
 }
 
 func validateContractChange(change Attribute) error {
-	validators := map[AttributeKey]func(string) error{
-		AttributeKeyName:       validateName,
-		AttributeKeyBaseImgURI: validateBaseImgURI,
-		AttributeKeyMeta:       validateMeta,
+	validators := map[string]func(string) error{
+		AttributeKeyName.String():       validateName,
+		AttributeKeyBaseImgURI.String(): validateBaseImgURI,
+		AttributeKeyMeta.String():       validateMeta,
 	}
 
 	return validateChange(change, validators)
 }
 
 func validateTokenClassChange(change Attribute) error {
-	validators := map[AttributeKey]func(string) error{
-		AttributeKeyName: validateName,
-		AttributeKeyMeta: validateMeta,
+	validators := map[string]func(string) error{
+		AttributeKeyName.String(): validateName,
+		AttributeKeyMeta.String(): validateMeta,
 	}
 
 	return validateChange(change, validators)
 }
 
-func validateChange(change Attribute, validators map[AttributeKey]func(string) error) error {
-	validator, ok := validators[AttributeKeyFromString(change.Key)]
+func validateChange(change Attribute, validators map[string]func(string) error) error {
+	validator, ok := validators[change.Key]
```

Comment:
Is there any reason to change like this? Overall, isn't it more consistent to use the key as the attribute key data type? I don't know the context of this changes so hope to understand more clearly.

Reply:
Using the keys makes the logic case insensitive. The return type of the conversion does not include any errors, but I didn't want to change the return types, because it's a deprecated module.